### ISSUE TITLE
TestAbilities discard cards very sluggish.

### ISSUE THREAD

**Issue by lodici** [magarena/magarena#1132]
Running TestAbilities, at the end of your first you have to discard a lot of cards. Each card click takes forever. Does the AI do any thinking on each card click? The thinking indicator does not appear so I assume not but then it begs the question why should discarding a card be so slow.

**Comment by melvinzhang** [comment]
No, the AI is not running when you are making a choice. Unless it is from the previous AI thinking step and a result of issue #1131 You can investigate using visualvm profiler to see which methods are taking up CPU cycles.

**Comment by lodici** [comment]
7a661ab has made a difference with a noticeable improvement in response.
